cloudinit/config/cc_power_state_change.py

```diff
@@ -4,67 +4,80 @@
 #
 # This file is part of cloud-init. See LICENSE file for license information.
 
-"""
-Power State Change
-------------------
-**Summary:** change power state
-
-This module handles shutdown/reboot after all config modules have been run. By
-default it will take no action, and the system will keep running unless a
-package installation/upgrade requires a system reboot (e.g. installing a new
-kernel) and ``package_reboot_if_required`` is true. The ``power_state`` config
-key accepts a dict of options. If ``mode`` is any value other than
-``poweroff``, ``halt``, or ``reboot``, then no action will be taken.
-
-The system
-can be shutdown before cloud-init has finished using the ``timeout`` option.
-The ``delay`` key specifies a duration to be added onto any shutdown command
-used. Therefore, if a 5 minute delay and a 120 second shutdown are specified,
-the maximum amount of time between cloud-init starting and the system shutting
-down is 7 minutes, and the minimum amount of time is 5 minutes. The ``delay``
-key must have an argument in either the form ``'+5'`` for 5 minutes or ``now``
-for immediate shutdown.
-
-Optionally, a command can be run to determine whether or not
-the system should shut down. The command to be run should be specified in the
-``condition`` key. For command formatting, see the documentation for
-``cc_runcmd``. The specified shutdown behavior will only take place if the
-``condition`` key is omitted or the command specified by the ``condition``
-key returns 0.
-
-.. note::
-    With Alpine Linux any message value specified is ignored as Alpine's halt,
-    poweroff, and reboot commands do not support broadcasting a message.
-
-**Internal name:** ``cc_power_state_change``
-
-**Module frequency:** per instance
-
-**Supported distros:** all
-
-**Config keys**::
-
-    power_state:
-        delay: <now/'+minutes'>
-        mode: <poweroff/halt/reboot>
-        message: <shutdown message>
-        timeout: <seconds>
-        condition: <true/false/command>
-"""
+"""Power State Change: Change power state"""
 
 import errno
 import os
 import re
 import subprocess
 import time
+from textwrap import dedent
 
 from cloudinit import subp, util
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.distros import ALL_DISTROS
 from cloudinit.settings import PER_INSTANCE
 
 frequency = PER_INSTANCE
 
 EXIT_FAIL = 254
 
+MODULE_DESCRIPTION = """\
+This module handles shutdown/reboot after all config modules have been run. By
+default it will take no action, and the system will keep running unless a
+package installation/upgrade requires a system reboot (e.g. installing a new
+kernel) and ``package_reboot_if_required`` is true.
+
+Using this module ensures that cloud-init is entirely finished with
+modules that would be executed, and avoids any error/log messages
```

Comment:
>... and avoids any error/log messages
that may go to the console as a result of system services like
syslog being taken down while cloud-init is running.

It sounds like adding this module solved a problem previously existed in cloud-init. This information seems valuable as a breadcrumb for future developers, but I doubt it is valuable / actionable for end users. Think we still need it?